### DISCUSSION TITLE
"Whole word only" in batch edit

### DISCUSSION THREAD

**Discussion by DrReddP** [SubtitleEdit/subtitleedit#5060] [Ideas]
I'm editing a series that has a ton of extra dashes, so I just removed them all in batch and am adding them back as need be. The issue is, in batch, there's no "whole word only" option like there is when you do multiple replace for a single file.
The issue here: I'm editing The Sopranos. They use the term "pre-med" a lot, but if I batch mult replace all the "premed"s, it's going to screw with the word "premeditated" as well. How do I tell mult replace in batch to only replace whole words?

![1](https://user-images.githubusercontent.com/82276985/119242696-221abf00-bb2e-11eb-981e-b2accb8ee738.png)

**Comment by OmrSi** [comment]
You can use the `Regular expression` option and write:
Find what: `\bpremed\b`
Replace with: `pre-med`

Speaking of Regular expression, you could have also replaced the dashes that are only in the the start of the line like this:
Find what: `^-`
Replace with: `nothing`

Or if you want to account for positioning and italic tags:
Find what: `(?<=^(?:{[^}]+})*(?:<[^>]*>)*)-`
Replace with: `nothing`

**Reply by DrReddP** [reply]
What on earth is a regular expression and where am I supposed to be typing these things

**Reply by OmrSi** [reply]
![image](https://user-images.githubusercontent.com/20923700/119243763-721b7500-bb72-11eb-91c2-e3c52b7f98d9.png)

**Reply by OmrSi** [reply]
![image](https://user-images.githubusercontent.com/20923700/119243782-8eb7ad00-bb72-11eb-9e72-c238e8b51290.png)

**Reply by OmrSi** [reply]
Note that Regular expressions are Case sensitive by default.

**Reply by DrReddP** [reply]
Thank you

**Reply by OmrSi** [reply]
You're most welcome.

**Reply by DrReddP** [reply]
When you say
Find what: ^-
Replace with: nothing

Do you mean actually type the word "nothing"?

**Reply by OmrSi** [reply]
No,  just leave it empty.
